rdtools/plotting.py

```diff
@@ -438,3 +438,67 @@ def availability_summary_plots(power_system, power_subsystem, loss_total,
     ax4.legend()
     ax4.set_ylabel('Cumulative Energy [kWh]')
     return fig
+
+
+def degradation_timeseries_plot(yoy_info, rolling_days=365, include_ci=True, **kwargs):
+    '''
+    Plot resampled time series of degradation trend with time
+
+    Parameters
+    ----------
+    yoy_info : dict
+        a dictionary with keys:
+
+        * YoY_values - pandas series of right-labeled year on year slopes
+        * renormalizing_factor - float value used to recenter data
+        * exceedance_level - the degradation rate that was outperformed with
+          a probability given by the ``exceedance_prob`` parameter in
+          the :py:func:`.degradation.degradation_year_on_year`
+    rolling_days: int
+        Number of days for rolling window. Note that the window must contain
+        at least 50% of datapoints to be included in rolling plot.
+    include_ci : boolean
+        calculate and plot 2-sigma confidence intervals along with rolling median
+        (true by default)
+    kwargs :
+        Extra parameters passed to plotting.degradation_summary_plots()
+
+    Note
+    ----
+    It should be noted that the yoy_rd, yoy_ci and yoy_info are the outputs
+    from :py:func:`.degradation.degradation_year_on_year`.
+
+    Returns
+    -------
+    matplotlib.figure.Figure
+    '''
+
+    def _bootstrap(x, percentile, reps):
+        # stolen from degradation_year_on_year
+        n1 = len(x)
+        xb1 = np.random.choice(x, (n1, reps), replace=True)
+        mb1 = np.nanmedian(xb1, axis=0)
+        return np.percentile(mb1, percentile)
+
+    try:
+        results_values = yoy_info['YoY_values']
+
+    except KeyError:
+        raise KeyError("yoy_info input dictionary does not contain key `YoY_values`.")
+
+    roller = results_values.resample('d').mean().rolling(rolling_days, min_periods=rolling_days//2)
```

Comment:
I’m not sure about `.mean()` here what’s the reasoning?

Reply:
I used `.resample('d').mean()` as a hacky way of converting an index with gaps/jumps to a continuous daily index, filling NaN for the missing values.  The point was to be able to define the window width with an integer instead of an offset.  An explicit `reindex()` is probably clearer, or maybe just use an offset in the first place.

Reply:
I found the following code to result in equivalent results if you don't like the resample:

`roller = results_values.rolling(f'{rolling_days}d', min_periods=rolling_days//2)`

data is 1:1 comparable with fewer NaN's in the beginning:
![image](https://user-images.githubusercontent.com/23244088/180268757-4218424c-63f7-4a5e-91d0-6d559ce9aadc.png)

Reply:
I'm not sure what's causing this nbval failure, it's occurring in a soiling calculation that I haven't changed at all...